c7n/resources/s3.py

```diff
@@ -1493,40 +1479,28 @@ def process(self, buckets):
 
     def process_bucket(self, bucket):
         s3 = bucket_client(local_session(self.manager.session_factory), bucket)
-        state = self.data.get('state', 'enable')
-        scopes = self.data.get('scope', ['All'])
+        config = {}
         try:
             config = s3.get_public_access_block(
                 Bucket=bucket['Name'])['PublicAccessBlockConfiguration']
         except ClientError as e:
             if e.response['Error']['Code'] == 'NoSuchPublicAccessBlockConfiguration':
-                # Set config to none because NoSuchPublicAccessBlockConfiguration error
-                # was returned. This is important later because it is symbolic of
-                # all blocks being set to false. See if/else statement below
-                config = None
+                # When there is a None config, set all to false to be symbolic
+                for key in self.keys:
```

Comment:
minor. its probably best to just initialize these values as false when creating the config dictionary, that removes these lines which are untested. 

```
config = {k: False for k in self.keys}
```

Reply:
Oh nice! I had played with list comprehensions already but dict comprehension is still relatively new  to me. I'll add in after investigating the bug reported.